jujugui/static/gui/src/app/components/unit-list/unit-list.js

```diff
@@ -84,6 +84,24 @@ YUI.add('unit-list', function() {
     },
 
     /**
+      Remove the selected units.
+
+      @method _handleRemoveUnits
+    */
+    _handleRemoveUnits: function() {
+      var units = [];
+      var refs = this.refs;
+      Object.keys(refs).forEach(function (ref) {
+        var isInstance = refs[ref] instanceof juju.components.UnitListItem;
```

Comment:
This check is fairly resource intensive and may cause perf issues in a large environment. Maybe instead we should set a naming pattern for refs which we could compare against... `UnitListItem-mysql/0` for example